crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -5985,7 +5985,11 @@ mod tests {
             "src/a.py",
             &["foo", "<listcomp>"],
             "x",
-            "@Todo(async iterables/iterators)",
+            if cfg!(debug_assertions) {
+                "@Todo(async iterables/iterators)"
+            } else {
+                "@Todo"
+            },
```

Comment:
An alternative here would be to rename `assert_scope_ty` to `assert_scope_ty_str`, and then introduce a new helper to directly assert on the actual `Type`, not its string representation. I would hope that this test can be migrated to an mdtest sooner or later (#13696).